clang/lib/CodeGen/CGStmtOpenMP.cpp

```diff
@@ -3044,6 +3046,9 @@ void CodeGenFunction::EmitOMPForOuterLoop(
   OuterLoopArgs.DKind = LoopArgs.DKind;
   EmitOMPOuterLoop(DynamicOrOrdered, IsMonotonic, S, LoopScope, OuterLoopArgs,
                    emitOMPLoopBodyWithStopPoint, CodeGenOrdered);
+  if (DynamicOrOrdered) {
```

Comment:
What has ordered to do with the dispatch (de)init?
We need DST for schedule dynamic, regardless of ordered, no?

Reply:
They are handled together all over the place and that is the same condition under which the init is emitted so it has to be the same for deinit.